plugins/scaffolder-backend/src/scaffolder/tasks/types.ts

```diff
@@ -44,6 +44,7 @@ export type DbTaskEventRow = {
 
 export type TaskSpec = {
   baseUrl?: string;
+  token?: string | undefined;
```

Comment:
Tbh we really would like to keep the secrets a bit more isolated. Our thinking is that we put it in a separate database column that gets cleared as soon as the task has been executed. We also have the need to store more tokens already, see e.g. #5214, so would like to create some structure around that too, probably with each token being keyed by some form of identifier.

What do you think about adding a `secrets` DB column with a JSON serialized `text` field where we for now store `{ identityToken: <token> }`?

Reply:
Sounds wise. PR updated.